File: packages/text-zoom/ios/Plugin/Classes/Options/SetZoomOptions.swift
import Foundation
import Capacitor

@objc public class SetZoomOptions: NSObject {
    let zoom: Double

    init(_ call: CAPPluginCall) throws {
        self.zoom = try SetZoomOptions.getZoomFromCall(call)
    }

    private static func getZoomFromCall(_ call: CAPPluginCall) throws -> Double {
        guard let zoom = call.getDouble("zoom") else {
            throw CustomError.zoomMissing
        }
        if zoom <= 0.0 {
            throw CustomError.zoomInvalid
        }
        return zoom
    }
}

File: packages/text-zoom/android/src/main/java/io/capawesome/capacitorjs/plugins/textzoom/classes/options/SetZoomOptions.java
package io.capawesome.capacitorjs.plugins.textzoom.classes.options;

import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.textzoom.classes.CustomExceptions;

public class SetZoomOptions {

    private final double zoom;

    public SetZoomOptions(PluginCall call) throws Exception {
        this.zoom = SetZoomOptions.getZoomFromCall(call);
    }

    public double getZoom() {
        return zoom;
    }

    private static double getZoomFromCall(PluginCall call) throws Exception {
        Double zoom = call.getDouble("zoom");
        if (zoom == null) {
            throw CustomExceptions.ZOOM_MISSING;
        }
        if (zoom <= 0.0) {
            throw CustomExceptions.ZOOM_INVALID;
        }
        return zoom;
    }
}
